source/upgrade/upgrading-mattermost-server.rst

```diff
@@ -1,44 +1,121 @@
+      For a complete breakdown of PostgreSQL queries, as well as their impact and duration, see the `Mattermost v6.0 DB Schema Migrations Analysis <https://gist.github.com/streamer45/59b3582118913d4fc5e8ff81ea78b055#postgresql-1>`__.
+
+Upgrading from Releases Older than v5.35
+----------------------------------------
+
+Customers upgrading from a release older than Mattermost v5.35 should expect extended downtime when upgrading to v6.0, due to the introduction of backend database architecture introduced in v5.35. This upgrade path isn't recommended for large installations. See the `Mattermost Changelog <https://docs.mattermost.com/install/self-managed-changelog.html?highlight=changelog#id39>`__ documentation for details.
+
+Upgrading from Releases Prior to v5.0
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you're upgrading from a version prior to v5.0, you can't upgrade directly to v6.0. Instead, we strongly recommend approaching the upgrade in phases, starting with an upgrade to the latest `Extended Support Release <https://docs.mattermost.com/administration/extended-support-release.html>`__ first, followed by the upgrade to v6.0.
+
+During the first phase of updates, you must modify your service file to work with the binary changes introduced with v5.0. Your execution directory should point to the Mattermost base directory (i.e. ``/opt/mattermost``), and your binary should point to the ``mattermost`` binary (i.e. ``/opt/mattermost/bin/mattermost``).
+
+Upgrading High Availability Deployments
+---------------------------------------
+
+In `High Availability <https://docs.mattermost.com/scale/high-availability-cluster.html>`__ environments, you should expect to schedule downtime for the upgrade to v6.0. Based on your database size and setup, the migration to v6.0 can take a significant amount of time, and may even lock the tables for posts which will prevent your users from posting or receiving messages until the migration is complete.
+
+Ensure you review the `High Availability Cluster Upgrade Guide <https://docs.mattermost.com/scale/high-availability-cluster.html#upgrade-guide>`__, as well as the :doc:`important-upgrade-notes` to make sure you're aware of any actions you need to take before or after upgrading from your particular version.
+
+.. important::
+
+  We only support one minor version difference between server versions when performing a rolling upgrade. For example v5.27.1 + v5.27.2 or v5.26.4 + v5.27.1 is supported, whereas v5.25.5 + v5.27.0 is not supported. 
```

Comment:
We support ESR to ESR though, don't we?  @agnivade @streamer45 ... BofA has done at least 1 or 2 ESR to ESR upgrades in a "Rolling" way.

Reply:
I think the official stance is that we don't support "rolling" upgrades for ESR to ESR.

We do _try_ to support it, but if it's necessary, we reserve the right to not make a rolling upgrade possible.